pallets/utxo/src/lib.rs

```diff
@@ -133,6 +133,50 @@ pub mod pallet {
         fn send_to_address(u: u32) -> Weight;
     }
 
+    /// Absolute time lock amount
+    #[derive(Eq, PartialEq, Clone, Copy, Encode, Decode, Debug, Default)]
+    #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+    pub struct AbsTimeLock(#[codec(compact)] u64);
+
+    impl AbsTimeLock {
+        // Values under this constant are considered to be a block number, time stamp otherwise.
+        // This constant is identical to what is used in Bitcoin.
+        // At 1min/block, the block numbers run out in about 900 years after the genesis block.
+        const THRESHOLD: u64 = 500_000_000u64;
+
+        pub fn from_u64(tl: u64) -> Self {
+            Self(tl)
+        }
+
+        pub fn from_blocks(num_blocks: u64) -> Option<Self> {
```

Comment:
Perhaps a better way to implement this, instead of these "from_x" functions, make a getter that returns an enum, which has the type inside it based on the threshold. Something like `get_concrete` vs `get_raw`. Imagine how this will be used in code.

Reply:
Yep. This way, the underlying type for bitcoin scripts is always u64. But for the developer who wants to use this struct, it can be distinguished, if so they please.